app/Helpers/database/forum.php

```diff
@@ -438,10 +439,18 @@ function authorizeAllForumPostsForUser(User $user): bool
 
 function isUserSubscribedToForumTopic(int $topicID, int $userID): bool
 {
-    return isUserSubscribedTo(
-        SubscriptionSubjectType::ForumTopic,
-        $topicID,
-        $userID,
-        "SELECT 1 FROM ForumTopicComment WHERE ForumTopicID = $topicID AND author_id = $userID"
-    );
+    $explicitSubcription = Subscription::where('subject_type', SubscriptionSubjectType::ForumTopic)
+        ->where('subject_id', $topicID)
+        ->where('user_id', $userID)
+        ->first();
+
+    if ($explicitSubcription) {
+        return $explicitSubcription->state == 1;
+    }
+
+    $implicitSubscription = ForumTopicComment::where('ForumTopicID', $topicID)
+        ->where('author_id', $userID)
+        ->exists();
+
+    return $implicitSubscription;
```

Comment:
nit: Seems like this can just `return` rather than needing a variable (might also be more clear, as `$implicitSubscription` isn't a subscription - it's a boolean):

```php
return ForumTopicComment::where('ForumTopicID', $topicID)
    ->where('author_id', $userID)
    ->exists();
```

Reply:
I wanted to describe why it was querying the ForumTopicComment table, and this seemed to pair better with the previous query. I guess a comment might be more appropriate.

Reply:
I like the comment more 👍